samlandfried/fair_bnb
# Reviews

#### What does this PR do?
This PR includes the feature for users to write reviews on a property show page
   - form uses ajax/jquery to render on page w/o reload
   - reviews are paginated w/ a max of 3 per page
   - average rating for properties is generated on show, index and home page

#### Where should the reviewer start?
   - feature/model tests are written for reviews
   - most of implementation happens on _reviews partial
   - logic behind creating a review happens in review_controller
   - javascript, ajax, jquery can be found in ratings.js

#### How should this be manually tested?
   - Create some reviews!
   - database is seeded with reviews so you can check if the average rating is correct

#### Any background context you want to provide?
  - n/a

#### What are the relevant tickets?
   [#146545459, #145637645, #145637757]

#### Screenshots (if appropriate)

<img width="989" alt="screen shot 2017-06-07 at 5 58 05 pm" src="https://user-images.githubusercontent.com/21000661/26906494-e9aedad0-4baa-11e7-8fbc-0dc87bf0a576.png">
<img width="987" alt="screen shot 2017-06-07 at 5 57 55 pm" src="https://user-images.githubusercontent.com/21000661/26906493-e9ad9c4c-4baa-11e7-80a0-48c5b6a3704d.png">
<img width="902" alt="screen shot 2017-06-07 at 5 57 30 pm" src="https://user-images.githubusercontent.com/21000661/26906495-e9b2f430-4baa-11e7-8c9a-5594905a66ab.png">


#### Questions:
  - Do Migrations Need to be ran? 
      - Yes
  - Do Environment Variables need to be set? 
      - No
  - Any other deploy steps?
      - No
